activerecord/lib/active_record.rb

```diff
@@ -405,6 +405,12 @@ def self.commit_transaction_on_non_local_return=(value)
   singleton_class.attr_accessor :migration_strategy
   self.migration_strategy = Migration::DefaultStrategy
 
+  ##
+  # :singleton-method: migration_generator_default_database
+  # Specify the default database to generate migrations for.
+  singleton_class.attr_accessor :migration_generator_default_database
+  self.migration_generator_default_database = "primary"
```

Comment:
> But no where else (except for single db applications) do we assume there is a config named "primary"

https://github.com/rails/rails/blob/9f1dec2ea5155205a880f6e6e232cf9ea6da2d8c/activerecord/lib/active_record/database_configurations.rb#L143

https://github.com/rails/rails/blob/9f1dec2ea5155205a880f6e6e232cf9ea6da2d8c/activerecord/lib/active_record/tasks/database_tasks.rb#L108

> I think this would be fine if it was set to nil and only used if it is set by the application, but it can't have a default value

Ok, happy to change it to `nil`, if that unblocks the PR.

> If there are more than the migration ones, this should be renamed to default_database_for_generators

Will check 👍🏻

Reply:
Those aren't assuming there is _definitely_ a primary, just if there is one, use it. But in general, it's first defined config. And that can be named anything.

Reply:
Turns out the [ModelGenerator](https://github.com/rails/rails/blob/3c6adf2994f091c5e52ec10719f05b12b609f715/activerecord/lib/rails/generators/active_record/model/model_generator.rb) could also be tweaked to use a _configurable default database_, so I went ahead with your suggestion, and renamed the configuration option to `default_database_for_generators`.

Let me know what you think.